api/nvidia.com/resource/gpu/v1alpha1/gpuconfig.go

```diff
@@ -68,8 +84,21 @@ func (c *GpuConfig) Normalize() error {
 
 // Validate ensures that GpuConfig has a valid set of values.
 func (c *GpuConfig) Validate() error {
-	if c.Sharing == nil {
-		return fmt.Errorf("no sharing strategy set")
+	if c.DriverConfig.Driver == NvidiaDriver {
+		if c.Sharing == nil {
+			return fmt.Errorf("no sharing strategy set")
+		}
+		if err := c.Sharing.Validate(); err != nil {
+			return err
+		}
+	} else {
+		if c.Sharing != nil {
+			return fmt.Errorf("sharing strategy cannot be provided while using non-nvidia driver")
+		}
 	}
-	return c.Sharing.Validate()
+	if err := c.DriverConfig.Validate(); err != nil {
+		return err
+	}
```

Comment:
Should we validate the driver config *before* validating the sharing config which depends on it?